plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt

```diff
@@ -375,50 +373,69 @@ class CodeWhispererCodeScanSession(val sessionContext: CodeScanSessionContext) {
     }
 
     fun mapToCodeScanIssues(recommendations: List<String>): List<CodeWhispererCodeScanIssue> {
-        val scanRecommendations: List<CodeScanRecommendation> = recommendations.map {
-            val value: List<CodeScanRecommendation> = MAPPER.readValue(it)
-            value
-        }.flatten()
+        val scanRecommendations = recommendations.flatMap { MAPPER.readValue<List<CodeScanRecommendation>>(it) }
         if (isProjectScope()) {
             LOG.debug { "Total code scan issues returned from service: ${scanRecommendations.size}" }
         }
-        return scanRecommendations.mapNotNull {
+        return scanRecommendations.mapNotNull { recommendation ->
             val file = try {
                 LocalFileSystem.getInstance().findFileByIoFile(
-                    Path.of(sessionContext.sessionConfig.projectRoot.path, it.filePath).toFile()
+                    Path.of(sessionContext.sessionConfig.projectRoot.path, recommendation.filePath).toFile()
                 )
             } catch (e: Exception) {
-                LOG.debug { "Cannot find file at location ${it.filePath}" }
+                LOG.debug { "Cannot find file at location ${recommendation.filePath}" }
                 null
             }
-            when (file?.isDirectory) {
-                false -> {
-                    runReadAction {
-                        FileDocumentManager.getInstance().getDocument(file)
-                    }?.let { document ->
-                        val endLineInDocument = min(max(0, it.endLine - 1), document.lineCount - 1)
+
+            if (file?.isDirectory == false) {
+                runReadAction {
+                    FileDocumentManager.getInstance().getDocument(file)
+                }?.let { document ->
+
+                    val documentLines = document.getText().split("\n")
+                    val (startLine, endLine) = recommendation.run { startLine to endLine }
+                    var shouldShow = true
+
+                    for (codeSni in recommendation.codeSnippet) {
+                        val lineNumber = codeSni.number - 1
+                        if (codeSni.number in startLine..endLine) {
+                            val documentLine = documentLines.getOrNull(lineNumber)
+                            if (documentLine != codeSni.content) {
+                                shouldShow = false
+                                break
+                            }
+                        }
+                    }
+
+                    if (shouldShow) {
```

Comment:
nit: shouldShow is vague name. change it to shouldDisplayIssue or something else.